seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/catalog/MilvusCatalog.java

```diff
@@ -0,0 +1,402 @@
+
+        if (CollectionUtils.isNotEmpty(tableSchema.getVectorIndexes())) {
+            createIndexInternal(tablePath, tableSchema.getVectorIndexes());
+        }
+    }
+
+    private void createIndexInternal(TablePath tablePath, List<VectorIndex> vectorIndexes) {
+        for (VectorIndex index : vectorIndexes) {
+            CreateIndexParam createIndexParam =
+                    CreateIndexParam.newBuilder()
+                            .withDatabaseName(tablePath.getDatabaseName())
+                            .withCollectionName(tablePath.getTableName())
+                            .withFieldName(index.getFieldName())
+                            .withIndexName(index.getIndexName())
+                            .withIndexType(IndexType.valueOf(index.getIndexType()))
+                            .withMetricType(MetricType.valueOf(index.getMetricType()))
+                            .build();
+
+            client.createIndex(createIndexParam);
+        }
+    }
+
+    public void createTableInternal(TablePath tablePath, CatalogTable catalogTable) {
+        try {
+            TableSchema tableSchema = catalogTable.getTableSchema();
+            List<FieldType> fieldTypes = new ArrayList<>();
+            for (Column column : tableSchema.getColumns()) {
+                fieldTypes.add(convertToFieldType(column, tableSchema.getPrimaryKey()));
+            }
+
+            Boolean enableDynamicField =
+                    (null != tableSchema.getEnableDynamicField())
+                            ? tableSchema.getEnableDynamicField()
+                            : config.get(MilvusSinkConfig.ENABLE_DYNAMIC_FIELD);
+
+            CreateCollectionParam.Builder builder =
+                    CreateCollectionParam.newBuilder()
+                            .withDatabaseName(tablePath.getDatabaseName())
+                            .withCollectionName(tablePath.getTableName())
+                            .withFieldTypes(fieldTypes)
+                            .withEnableDynamicField(enableDynamicField)
+                            .withConsistencyLevel(ConsistencyLevelEnum.BOUNDED);
+            if (null != catalogTable.getComment()) {
+                builder.withDescription(catalogTable.getComment());
+            }
+
+            CreateCollectionParam createCollectionParam = builder.build();
+            R<RpcStatus> response = this.client.createCollection(createCollectionParam);
+            if (!Objects.equals(response.getStatus(), R.success().getStatus())) {
+                throw new CatalogException(
+                        String.format("Create collection failed, err=%s", response.getMessage()),
+                        response.getException());
+            }
+        } catch (Exception e) {
+            throw new CatalogException(
+                    String.format("Failed creating collection %s", tablePath.getFullName()), e);
+        }
+    }
+
+    private FieldType convertToFieldType(Column column, PrimaryKey primaryKey) {
```

Comment:
create `MilvustypeConverter.java`

move `convertToFieldType` and `convertToDataType` into `MilvustypeConverter`



reference
https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/psql/PostgresTypeConverter.java

Reply:
I've MilvusConvertUtils, and have move convertToDataType to MilvusConvertUtils. convertToFieldType contains options, can't move.